jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarProcessors.java

```diff
@@ -34,10 +42,17 @@
    * @param mode processing mode
    * @return JarProcessor
    * @throws IOException if I/O error occurs when opening the jar file
+   * @throws IncompatibleBaseImageJavaVersionException if the base image java version cannot support
+   *     this build
    */
   public static JarProcessor from(
       Path jarPath, TempDirectoryProvider temporaryDirectoryProvider, ProcessingMode mode)
-      throws IOException {
+      throws IOException, IncompatibleBaseImageJavaVersionException {
+    Integer jarVersion = getVersion(jarPath);
+    if (jarVersion > 11) {
+      throw new IncompatibleBaseImageJavaVersionException(11, jarVersion);
```

Comment:
In build plugins, we catch this exception at a higher stack and provide a detailed error message with an action item. Maybe we can throw an `IllegalStateException`? The message should report the app version, what the problem is, and how to fix.